### DISCUSSION TITLE
Gun detection

### DISCUSSION THREAD

**Discussion by kennynakamura** [sepinf-inc/IPED#673] [Ideas]
Hello, me again!  Are you all right?
I was studying about image recognition, and I remembered face_recognition task in IPED. 
It would be interesting a task to recognize guns? 
I have the python script of that task, just having some trouble in put in IPED. 
But I already foresee some problem regarding processing and maybe will need a configuration like face_recognition, would it be worth it for this function?

**Comment by lfcnassif** [comment]
Hi @kennynakamura, everything is fine, thanks. I think this would be a great addition, thank you for your willingness to contribute again! I can try to help you with this after I return from vacation next week.

**Reply by kennynakamura** [reply]
Ok! Later I post the commented script.
I'm happy to be able to try to help! Enjoy your vacation!

**Reply by kennynakamura** [reply]
Here is the link I was inspired by for this idea. Is a paylink, but if you open it in an incognito window, you can see it without any problems.

https://medium.com/@quangnhatnguyenle/how-to-train-yolov3-on-google-colab-to-detect-custom-objects-e-g-gun-detection-d3a1ee43eda1

And here is the code I adapted, he needs some files for work. 
https://github.com/kennynakamura/GunDetectionTask/blob/main/GUNdetectionTask.py
https://drive.google.com/drive/folders/1ub9A46AyRMU1781gsaMUmhP1EhI3ffTt
It doesn't work very well, some C++ problems, I don't know exactly what could be happening.  but I believe it's a problem with openCV lib.

Here is a code in python for Command line , the flag part is very rudimentary,
https://github.com/kennynakamura/GunDetectionTask/blob/main/GUNdetection.py

This "ACERTO" and "ERRO" are some tests I did
![gundetection](https://user-images.githubusercontent.com/8608969/128360222-f3d88d9f-1c78-4ba5-893d-c1e48543b1cc.png)

**Reply by kennynakamura** [reply]
Hi, I was looking and I saw that there is no dataset for images of rifles. So I collected images and trained, so the model I trained has the possibility to detect pistol and rifle. This is the link for a colab notebook, I don't know if you've heard of it. It's intuitive how it works, but if you have any questions, I'm here to help. It has the automatic download link for the necessary files

https://colab.research.google.com/drive/1KJckreFWj5GcjUUy1GMjcfLjE402c6Jf?usp=sharing

**Reply by lfcnassif** [reply]
Thanks @kennynakamura. Actually for now I have some issues (and PRs for review) in my roadmap, not sure about other devs, but this for sure would be a great addition. When this is targeted, I think maybe it would be a good idea to train a multiclass classifier (guns, money, drugs, documents, people, etc).
